src/clustering/administration/servers/auto_reconnect.cc

```diff
@@ -87,7 +87,18 @@ void auto_reconnector_t::try_reconnect(const server_id_t &server,
     exponential_backoff_t backoff(50, 15 * 1000);
     try {
         while (!interruptor.is_pulsed()) {
-            connectivity_cluster_run->join(last_known_address, join_delay_secs);
+            join_result_t result =
+                connectivity_cluster_run->join_blocking(
+                    last_known_address, boost::none, server,
+                    join_delay_secs,
+                    auto_drainer_t::lock_t(&drainer));
```

Comment:
@danielmewes this is causing a guarantee error:
```
2016-06-24T11:16:51.422715984 708.920404s error: Error in src/concurrency/auto_drainer.cc at line 74:
2016-06-24T11:16:51.422848529 708.920537s error: Guarantee failed: [p == parent]
2016-06-24T11:16:51.422891086 708.920579s error: Backtrace:
2016-06-24T11:16:51.863289243 709.360982s error: Fri Jun 24 11:16:51 2016
1 [0xa93e50]: backtrace_t::backtrace_t() at ??:?
2 [0xa941e3]: format_backtrace(bool) at ??:?
3 [0xa705bd]: report_fatal_error(char const*, int, char const*, ...) at ??:?
4 [0xa6ce45]: auto_drainer_t::lock_t::assert_is_holding(auto_drainer_t*) const at ??:?
5 [0xac8511]: connectivity_cluster_t::run_t::join_blocking(peer_address_t, boost::optional<peer_id_t>, boost::optional<server_id_t>, int, auto_drainer_t::lock_t) at ??:?
6 [0xc6a7ae]: auto_reconnector_t::try_reconnect(server_id_t const&, auto_drainer_t::lock_t) at ??:?
7 [0xc6aee7]: callable_action_instance_t<std::_Bind<std::_Mem_fn<void (auto_reconnector_t::*)(server_id_t const&, auto_drainer_t::lock_t)> (auto_reconnector_t*, server_id_t, auto_drainer_t::lock_t)> >::run_action() at ??:?
8 [0x7733f7]: coro_t::run() at ??:?
```

any idea why? not very familiar with your `auto_drainer_t`